### DISCUSSION TITLE
Key remapping limitation

### DISCUSSION THREAD

**Discussion by romaia** [helix-editor/helix#5339] [Q&A]
This is a somewhat similar question to #5234, but I would like to further expand it:

I would like to remap the key `*` to a similar behaviour that I have in vim, where the whole word under the cursor is searched.

Of course I could use `miw` as suggested there, but ideally, I really want to make `*` behave exactly as `miw`, But It does not seem that is possible.

I see in the source code that `mi` maps to `select_textobject_innter`:

```rust

        "m" => { "Match"
            "m" => match_brackets,
            "s" => surround_add,
            "r" => surround_replace,
            "d" => surround_delete,
            "a" => select_textobject_around,
            "i" => select_textobject_inner,
        },
```

So this means that the last `w` is handled inside of the `select_textobject_inner` handler.

It is possible to achive such behaviour? Is this a know limitation? Or can I somehow send that missing `w` to `select_textobject_inner`?

Another idea would be to allow creation of "macros', where I could configure `"*" = "miw"` somehow.

**Comment by kirawi** [comment]
https://github.com/helix-editor/helix/pull/4709
